packages/core/src/modules/indy/services/IndyVerifierService.ts

```diff
@@ -4,35 +4,64 @@ import { Lifecycle, scoped } from 'tsyringe'
 
 import { AgentConfig } from '../../../agent/AgentConfig'
 import { IndySdkError } from '../../../error'
+import { isIndyError } from '../../../utils/indyError'
+import { IndyLedgerService } from '../../ledger/services/IndyLedgerService'
 
 @scoped(Lifecycle.ContainerScoped)
 export class IndyVerifierService {
   private indy: typeof Indy
+  private ledgerService: IndyLedgerService
 
-  public constructor(agentConfig: AgentConfig) {
+  public constructor(agentConfig: AgentConfig, ledgerService: IndyLedgerService) {
     this.indy = agentConfig.agentDependencies.indy
+    this.ledgerService = ledgerService
   }
 
   public async verifyProof({
     proofRequest,
     proof,
     schemas,
     credentialDefinitions,
-    revocationRegistryDefinitions = {},
-    revocationStates = {},
   }: VerifyProofOptions): Promise<boolean> {
     try {
+      const { revocationRegistryDefinitions, revocationRegistryStates } = await this.getRevocationRegistries(proof)
+
       return await this.indy.verifierVerifyProof(
         proofRequest,
         proof,
         schemas,
         credentialDefinitions,
         revocationRegistryDefinitions,
-        revocationStates
+        revocationRegistryStates
       )
     } catch (error) {
-      throw new IndySdkError(error)
+      throw isIndyError(error) ? new IndySdkError(error) : error
+    }
+  }
+
+  private async getRevocationRegistries(proof: Indy.IndyProof) {
+    const revocationRegistryDefinitions: Indy.RevocRegDefs = {}
+    const revocationRegistryStates: Indy.RevStates = {}
+    for (const identifier of proof.identifiers) {
+      const revRegId = identifier.rev_reg_id
+      const timestamp = identifier.timestamp
+
+      //Fetch Revocation Registry Definition if not already fetched
+      if (revRegId && !revocationRegistryDefinitions[revRegId]) {
+        const { revocRegDef } = await this.ledgerService.getRevocationRegistryDefinition(revRegId)
+        revocationRegistryDefinitions[revRegId] = revocRegDef
+      }
+
+      //Fetch Revocation Registry by Timestamp if not already fetched
+      if (revRegId && timestamp && !revocationRegistryStates[revRegId]?.[timestamp]) {
+        if (!revocationRegistryStates[revRegId]) {
+          revocationRegistryStates[revRegId] = {}
+        }
+        const { revocReg } = await this.ledgerService.getRevocationRegistry(revRegId, timestamp)
+        revocationRegistryStates[revRegId][timestamp] = revocReg
```

Comment:
It's complaining about prototype pollution and that we shouldn't create an object prone to prototype polution

Reply:
I had noticed this as well but had initially thought it was something specific to my environment. I'll look to reproduce.

Reply:
Interestingly I wasn't able to recreate this--I wonder if it was resolved with our last types PR addition?